### PR TITLE
Ovftool import

### PR DESCRIPTION
**Pull request by daxgames**
This is a big change, not sure you ever looked it it based on my last email to you.  I never saw a response so I never did a pull request until now.

This uses OVFTool to do the import.  No clone required, No wait required, results in a more ESXi compatible VM Template that uses a managed vmdk type like seSparse, eagerZeroedThick, thin, or
thick when using the resulting artifact in ESXi as a template.  Will require some changes to  Packer Templates.

### CODE REVIEW

**Comment by lamdor** [comment]
@daxgames This looks like a great update on first glance. I'm going to run it against our internal packer build script to make sure it all works well and if so, I'll merge it in.

Thanks for the PR!

**Comment by daxgames** [comment]
Hopefully it works for you.  It works well here.

**Comment by lamdor** [comment]
I really tried to make this work (merge in progress at https://github.com/Banno/packer-post-processor-vsphere-ova/tree/merge-daxgames-ovftool_import), but I unfortunately kept getting `Transfer Failed, Error: Lost connection to target` when it was using ovftool to upload. :disappointed: 

I'm going to take a look and see if it works with the master branch so we can determine if it's a problem with our vSphere and not ovftool.

**Comment by daxgames** [comment]
Interesting.  Could it be certificate related?  Are you using trusted or self signed certs in vsphere?

**Comment by daxgames** [comment]
If self-signed you may need to add "insecure": "true" to your packer template post-processor section.  Not sure if this was in the readme changes, we could add it if not.

**Comment by daxgames** [comment]
This still uses SSL it just disabled certificate verification.

**Comment by daxgames** [comment]
I also noticed that my pull request changes your tabs from a 4 char tab to 2 spaces.  I can fix that if you like, just let me know.  It is our best practice to use 2 spaces vs tabs so my editor does that automatically.

**Comment by lamdor** [comment]
@daxgames Yeah, it's getting pass SSL for sure. It goes for about 30 seconds and then just stops. I can actually see the uploade vmx and vmdk within the datastore for a short while until it fails.

Re the formatting, I've used used gofmt to always reformat the file on save. Whether it's a tab of 2 spaces, I'm not terrible concerned. Just consistency is good for me.

**Comment by daxgames** [comment]
I am not sure I can even test it now.  Let me see what I can do and I will let you know.

**Comment by daxgames** [comment]
@rubbish, @Vikas027 got this to work, see issue: #10 

Can you both provide the anonymized relevant portion of your packer json template?  So we can try and discover why it works for one of you but not both.

**Comment by vikas027** [comment]
@rubbish  Here is my `post-processor` built from [here](https://github.com/daxgames/packer-post-processor-vsphere-ova/tree/ovftool_import).

``` json
  "post-processors": [
    {
      "type": "vsphere-ova",
      "host": "{{ user `vcenter_host` }}",
      "username": "{{ user `vcenter_user` }}",
      "password": "{{ user `vcenter_password` }}",
      "datacenter": "MyDC",
      "datastore": "datastore01",
      "vm_folder": "packer-templates",
      "vm_network": "VM Network",
      "virtual_hardware_version": "8",
      "remove_floppy": true,
      "remove_optical_drive": false,
      "disk_mode": "thin",
      "cluster": "MyCluster",
      "output_artifact_type": "template",
    }
  ]
```

@daxgames I also have two feature requests for your code.
- Ask the user for the OS Type, so that can be changed in the VM Template. As of now, it takes `Other (32-bit)` by default for all OS, which is not good.
- Add CDROM in the resultant template

Obviously, second one is not critical but first one is. I can use [shell-local](https://www.packer.io/docs/post-processors/shell-local.html) using this [script](https://gist.github.com/vikas027/388cc2047fc07d5b4e9e) and do the same, but it would be great to have everything packaged together. I am not well versed with go, otherwise I would have done this myself.

**Comment by daxgames** [comment]
@vikas027, Regarding your requests.

I hate to make the process interactive.  What if you could add it to the post-processor definition in the json file.  Not sure my code is what is doing this and not saying I can do it but I will look.

**Comment by vikas027** [comment]
@daxgames yeah, I meant an addition in Json template. Sorry, I wasn't clear. I too do not like user interactive things in my automation framework.

**Comment by vikas027** [comment]
@daxgames I have one more request for you to make your work production ready :), and this I believe would be an easy one.

For now, we cannot choose the ESXi host where we want to put the resultant template as the `ovftool` command ends with

``` sh
vi://<vcenter_user>:<vcenter_password>@<vcenter_host>/<datacenter>/host/<cluster>
```

It would be great if there is an option to choose among multiple ESXi hosts as well. To make that happen, the packer post-processor would have to append the ESXi host at the end

``` sh
vi://<vcenter_user>:<vcenter_password>@<vcenter_host>/<datacenter>/host/<cluster>/<esxi_host>
```

To sum up, I have three requests for you in order of priority
1. Option to choose the ESXi host (as explained above)
2. Option to choose OS Type, so that it can be changed in the VM Template. As of now, it takes `Other (32-bit)` by default for all OS, which is not good
3. Add CDROM in the resultant template  (not important at all)

Thank you in advance for you work on this.

@rubbish Not sure, why this didn't worked for you, I have checked this PR in four different environments, and it worked fine in all of them. I do not no Golang, but I would be happy to help in whatever way I can. Thanks.

**Comment by jmahowald** [comment]
Just giving a +1 in the sense that with this branch I was finally able to make a template in vcenter.

**Comment by lamdor** [comment]
I'll come back to this and retry it here in a bit. Not exactly sure what the problem I was running into.

**Comment by daxgames** [comment]
I made some changes to this today to fix some bugs.  Only fixes, no enhancements yet.

**Comment by daxgames** [comment]
Still working on it.

**Comment by mrjcleaver** [comment]
Still still? :)

**Comment by daxgames** [comment]
@mrjcleaver ???

**Comment by mrjcleaver** [comment]
Are you still working on it? 

I'm wondering if / when this will get get merged.

**Comment by daxgames** [comment]
@mrjcleaver I am not working on it.  As far as I am concerned it is complete.  

There are things people have asked for but I have no way to implement/test at this point.  

As for merge that is up to those that maintain this repo.
